src/object/cli_obj.c

```diff
@@ -1752,7 +1750,8 @@ check_query_flags(daos_obj_id_t oid, uint32_t flags, daos_key_t *dkey,
 /* check if the obj request is valid */
 static int
 obj_req_valid(tse_task_t *task, void *args, int opc, struct dtx_epoch *epoch,
-	      uint32_t *pm_ver, struct dc_object **p_obj, bool check_existence)
+	      uint32_t *pm_ver, struct dc_object **p_obj, bool spec_shard,
```

Comment:
Seems like spec_shard and check_existence are only potentially true from the call in dc_obj_fetch_task and are only needed for the DAOS_OBJ_RPC_FETCH case. Why not just inspect args.extra_flags in obj_req_valid() instead? It's already receiving args as a parameter. Then would be able to remove 2 params from obj_req_valid.